src/components/details-page/components/details-page/AccessRequestButton.tsx

```diff
@@ -0,0 +1,71 @@
+import Button from '@fellesdatakatalog/button';
+import React from 'react';
+import translations from '../../../../lib/localization';
+import { AccessRequest } from '../../../../types';
+import SC from './styled';
+import {
+  EventAction,
+  EventCategory,
+  trackSiteImproveEvent
+} from '../../../analytics-siteimprove/utils';
+
+const getAppliactionLink = async (location: string): Promise<string> => {
+  const accessRequestApi = `https://access-request.api.staging.fellesdatakatalog.digdir.no`;
+
+  const response = await fetch(
+    `${accessRequestApi}/access-request/${translations._language}${location}`
+  );
+
+  return response.text();
+};
+
+export function AccessRequestButton({
+  accessRequest,
+  entityId,
+  accessRequestUrl
+}: {
+  accessRequestUrl: string | undefined;
+  accessRequest: AccessRequest | undefined;
+  entityId: string | undefined;
+}) {
+  const trackAccessRequest = () => {
+    trackSiteImproveEvent({
+      category: EventCategory.DETAILS_PAGE,
+      action: EventAction.REQUEST_ACCESS,
+      label: entityId
+    });
+  };
+
+  const isEnturDataset = entityId === '8b285b05-dd33-31db-a1b6-0cf605bf05dd';
+
+  if (accessRequestUrl === 'soknad.kudaf.no' || isEnturDataset) {
```

Comment:
Litt usikker på om `soknad.kudaf.no` er riktig her, men brukte det fordi det er det som står [her](https://app.mural.co/t/entur7578/m/entur7578/1727254267103/76e888a39214a4a312c34cef1bac20cf98bc7196?wid=0-1727866772637). `https://frontend-soknad.paas2.uninett.no` er domene der søknaden ligger. Kanskje det er bedre accessRequestUrl?

Reply:
https://soknad.kudaf.no ser ut til å ikke være i bruk per nå og blir mest sannsynlig endret senere. Men siden vi ikke har requestAccessUrl definert på datasettet/api foreslår jeg følgende:

Når man kommer på detaljsiden gjør man en spørring til request-access-api hvor man spør om request-access url/info  for en gitt resurs (dette er konfigurert i request-access-api i en properties fil per nå). Hvis du får null tilbake blir request-access button ikke vist, hvis du får gyldig response viser man knappen med riktig window.location.href. Med andre ord man kan fjerne requestAccessUrl prop.

Reply:
Endepunktet på `access-request-api` oppretter en søknad hos kudaf, så har ikke lyst til å kalle det hver gang jeg er på en side. Det går alltids an å lage et nytt endepunkt som kun henter `requestAccessUrl`, men per nå bruker vi samme kall backend for å hente data som blir brukt frontend, så man vil ikke ha tilgang til noe ny data. 

Løste dette med å legge et nytt innslag i white-listen.